onnxruntime/contrib_ops/cuda/bert/group_query_attention_impl.cu

```diff
@@ -614,6 +614,31 @@ Status LaunchSeqlensToPosIds(contrib::GroupQueryAttentionParameters& parameters,
   return CUDA_CALL(cudaGetLastError());
 }
 
+__global__ void AllZeroesKernel(int* tensor, int size, bool* all_zeroes) {
+  int tid = blockDim.x * blockIdx.x + threadIdx.x;
+  if (tid == 0) {
+    *all_zeroes = true;
+  }
+  __syncthreads();
+  if (tid < size) {
+    if (tensor[tid] != 0) {
+      *all_zeroes = false;
+    }
+  }
+}
+
+Status AllZeroes(bool& all_zeroes, const int* tensor, int size, cudaStream_t stream, const int max_threads_per_block) {
```

Comment:
Calling an extra cuda kernel will be slower (comparing before/after this change).

How about keeping total_seq_len input, and use total_seq_len==seq_len to check whether it is prompt or not, and not use it in other places?